server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
Actually it's great that this is moved here, since it'd be a failure if there were no targets metadata and the user did not send target metadata.

Should the error be different?  It's not loading just for snapshot generation, since presumably since if there's no targets file available that should be a problem with the repo.